src/controllers/messages.controller.ts

```diff
@@ -0,0 +1,208 @@
+import { messagesDB } from '../models/db.index';
+import { RequestHandler } from 'express';
+import createHttpError from 'http-errors';
+
+export const getMessages: RequestHandler = async (req, res, errFunc) => {
+    try {
+        const messages = await messagesDB.findAll();
+        res.status(200).json(messages);
+    } catch (error) {
+        errFunc(error);
+    }
+};
+export const getLastMessages: RequestHandler = async (req, res, errFunc) => {
+    try {
+        const numMsgs = (req.params.numMsgs as unknown as number) || 10; //cast to number, or default to 10
+        const messages = await messagesDB.findAll({
+            limit: numMsgs,
+            order: [['time', 'DESC']],
+        });
+        res.status(200).json(messages);
+    } catch (error) {
+        errFunc(error);
+    }
+};
+
+export const getMessage: RequestHandler = async (req, res, errFunc) => {
+    const id = req.params.id;
+
+    try {
+        const message = await messagesDB.findByPk(id);
+
+        if (!message) {
+            throw createHttpError(400, 'Message not found');
+        }
+
+        res.status(200).json(message);
+    } catch (error) {
+        errFunc(error);
+    }
+};
+
+interface CreateMessageBody {
+    id: string,
+    nodeID?: string,
+    plotID?: string,
+    time?: Date,
+    battery?: string,
+    soilConductivity?: number,
+    humidity?: number,
+    temperature?: number,
+    sunlight?: number,
+}
+
+export const createMessage: RequestHandler<unknown, unknown, CreateMessageBody, unknown> = async (req, res, errFunc) => {
+    const id = req.body.id;
+    const nodeID = req.body.nodeID;
+    const plotID = req.body.plotID;
+    const time = req.body.time;
+    const battery = req.body.battery;
+    const soilConductivity = req.body.soilConductivity;
+    const humidity = req.body.humidity;
+    const temperature = req.body.temperature;
+    const sunlight = req.body.sunlight;
```

Comment:
lets not instantiate all of these. we have the type defined. to get that to be used lets do
```ts
const body = req.body as CreateMessageBody;
```

Reply:
second thought, lets call this machine or new machine instead of body